server/testdriver/junit4/src/main/java/org/infinispan/server/test/junit4/InfinispanXSiteServerTestMethodRule.java

```diff
@@ -0,0 +1,88 @@
+package org.infinispan.server.test.junit4;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.counter.api.CounterManager;
+import org.infinispan.server.test.api.HotRodTestClientDriver;
+import org.infinispan.server.test.api.RestTestClientDriver;
+import org.infinispan.server.test.api.TestClientXSiteDriver;
+import org.infinispan.server.test.core.TestClient;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import net.spy.memcached.MemcachedClient;
+
+/**
+ * @author Gustavo Lira &lt;glira@redhat.com&gt;
+ * @since 12.0
+ **/
+public class InfinispanXSiteServerTestMethodRule implements TestRule, TestClientXSiteDriver {
+   private final List<TestClient> testClients;
```

Comment:
this would make sense to be an Map.

Reply:
I thought you said to use List instead of Map here, I will revert

Reply:
you mean this: https://github.com/infinispan/infinispan/pull/8614#discussion_r482137476
it isn't in this class :)